src/pybind/mgr/volumes/fs/subvolspec.py

```diff
@@ -86,17 +86,23 @@ def trash_dir(self):
         """
         return os.path.join(self.subvolume_prefix.encode('utf-8'), b"_deleting")
 
-    def make_subvol_snap_path(self, snapdir, snapname):
+    def make_subvol_snap_path(self, snapdir, snapname=None):
         """
-        return the subvolume snapshot path for a given snapshot name
+        return the subvolume snapshot path for a given snapshot name or return the snapdir path
         """
-        return os.path.join(self.subvolume_path, snapdir, snapname)
+        if snapname is not None:
+            return os.path.join(self.subvolume_path, snapdir, snapname)
+        else:
+            return os.path.join(self.subvolume_path, snapdir)
 
-    def make_group_snap_path(self, snapdir, snapname):
+    def make_group_snap_path(self, snapdir, snapname=None):
         """
-        return the group snapshot path for a given snapshot name
+        return the group snapshot path for a given snapshot name or return the snapdir path
```

Comment:
write a separate method to return group's snapdir path and call it make_group_snapdir_path